tests/complex.nim

```diff
@@ -87,3 +87,16 @@ suite "Complex tests":
     rnw.setPosition(0)
     dsmo.serialize(rnw)
     check(true)
+
+  test "Periodic dynamic variant":
+    serializable:
+      type
+        VariantType = object
+          case a: uint8
+          of 1:
+            b: string
+          else:
+            c: uint8
+        PeriodicContainer = object
+          a: seq[VariantType]
+    check(true)
```

Comment:
I think we need a serialization test here in addition to the compilation test

Reply:
I'll try to add one when I get time again. The code change affects all variants, so serialization works in at least the cases that are already tested.

Reply:
I went ahead and added a basic test.